api/v1alpha1/healthcheck_types.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// HealthCheck defines the health check configuration.
+type HealthCheck struct {
+	// Timeout defines the time to wait for a health check response.
+	//
+	// +kubebuilder:validation:Format=duration
+	Timeout metav1.Duration `json:"timeout"`
+
+	// Interval defines the time between health checks.
+	//
+	// +kubebuilder:validation:Format=duration
+	Interval metav1.Duration `json:"interval"`
+
+	// UnhealthyThreshold defines the number of unhealthy health checks required before a backend host is marked unhealthy.
+	//
+	// +kubebuilder:validation:Minimum=1
+	UnhealthyThreshold uint32 `json:"unhealthyThreshold"`
+
+	// HealthyThreshold defines the number of healthy health checks required before a backend host is marked healthy.
+	//
+	// +kubebuilder:validation:Minimum=1
+	HealthyThreshold uint32 `json:"healthyThreshold"`
+
+	// HealthChecker defines the concrete health checker to do health checking.
```

Comment:
we need a better name here, it currently looks like
```
healthCheck:
  healthChecker:
  .....
```

Reply:
Thoughts on endpoint ?

Reply:
I feel a little bit confused.  Any thoughts from others?

Reply:
how about removing `healthChecking` and copying the fields directly into top level struct ?

Reply:
How about this ?

```yaml
  healthChecking: # optional
      connectionTimeout: duration # optional
      healthyCheckInterval: duration # optional, default=5s
      unhealthyCheckInterval: duration, # optional, default=10s
      unhealthyThreshold: int # optional, default=3
      healthyThreshold: int # optional, default=1
      type: Enum (grpc/http)
      grpc: # required when type=grpc
        upstreamName: string # required
        authority: string # optional
      http: # required when type=http
        expectedStatuses: # required, minItems=1
        - min: int # min=100, max=599
           max: int # min=100, max=599
        path: string # optional
        hostname: string # optional
```

Reply:
yup looking good !
* prefer "healthCheck" as the top level noun
* lets only support tcp and http for now, and add support for grpc later

Reply:
inline heathChecker to heathCheck and remove grpc checker.

done